### ISSUE TITLE
Hashing under SQLA needs improvement

### ISSUE THREAD

**Issue by szoupanos** [aiidateam/aiida_core#1771]
For example when hashing nodes, I get exceptions for the UUID (it is not hashable etc).
It seems that for the moment that a small fix like to following does the job.

```
diff --git a/aiida/common/hashing.py b/aiida/common/hashing.py
index c1bee46..2f4b80c 100644
--- a/aiida/common/hashing.py
+++ b/aiida/common/hashing.py
@@ -261,6 +261,11 @@ def _(object_to_hash, **kwargs):
 def _(object_to_hash, **kwargs):
     return make_hash_with_type('d', str(object_to_hash))

+import uuid
+@make_hash.register(uuid.UUID)
+def _(object_to_hash, **kwargs):
+    return make_hash_with_type('u', str(object_to_hash))
+
 @make_hash.register(Folder)
 def _(folder, **kwargs):
     # make sure file is closed after being read
```

**Comment by giovannipizzi** [comment]
question for @greschd: is the aim to have also the same hash between different db implementations? (sqla, django)
in this case, probably we should hash the uuid in the same way as django (string). otherwise, using a type 'u' is ok?

**Comment by giovannipizzi** [comment]
question for @greschd: is the aim to have also the same hash between different db implementations? (sqla, django)
in this case, probably we should hash the uuid in the same way as django (string). otherwise, using a type 'u' is ok?

**Comment by greschd** [comment]
Well, I think if we can easily make them be the same that's nice, but it's not a hard requirement. After all, if you change DB backend I think you can expect that the caches will be invalidated. For this purpose, you can always re-hash nodes if that's needed.

**Comment by szoupanos** [comment]
I understand that there is nothing against this change. So will proceed and add it to v0.12.2

**Comment by sphuber** [comment]
Fixed in PR #1861
